gameplay/src/JoystickControl.h

```diff
@@ -171,13 +207,35 @@ class JoystickControl : public Control
      */
     JoystickControl(const JoystickControl& copy);
 
-    float _radius; 
+    void getRegion(Vector2* regionCoordPairOut, const char* regionPropertyId) const;
+
+    Vector2 getPixelSize(const Vector2* regionCoordPair) const;
+
+    Vector2 getPixelSize(const Theme::ThemeImage* image) const;
+
+    Theme::ThemeImage * getNonEmptyImage(const char* id, Control::State state);
+
+    void updatePixelValues();
+
+    enum CoordPair
+    {
+        VALUE,
+        PERCENTAGE_STATE,
+        SIZE
+    };
+
+    float _radiusCoord[CoordPair::SIZE];
```

Comment:
I think we should instead  implement this by expanding the existing BOUNDS_BITS_xxxx
 to be PERCENTAGE_BIT_BOUNDS_X. You can then adding in PERCENTAGE_BITS_RADIUS and not need this enum.